atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/metadata/SchemaMetadataService.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.schema.metadata;
+
+import java.util.Map;
+import java.util.Optional;
+
+import com.palantir.atlasdb.schema.SchemaMetadata;
+
+public interface SchemaMetadataService {
```

Comment:
Can we have javadocs for this? It's mostly self-explanatory, but `decommissionSchema` was a bit unexpected and ambiguous.

Reply:
OK, added JavaDocs. Separately I've renamed `decommissionSchema` to `deleteSchemaMetadata` (and the associated tests).